addons/delivery/models/delivery_carrier.py

```diff
@@ -47,10 +32,30 @@ class DeliveryCarrier(models.Model):
     # Internals for shipping providers #
     # -------------------------------- #
 
-    delivery_type = fields.Selection([('grid', 'Default Delivery')], string="Delivery Type", default='grid', required='True')
+    name = fields.Char(string='Delivery Method', required=True, translate=True)
+    sequence = fields.Integer(help="Determine the display order", default=10)
+    delivery_type = fields.Selection([('fixed', 'Fixed Price'), ('base_on_rule', 'Based on Rule')], string='Price Computation', default='fixed', required='True')
 
+    partner_id = fields.Many2one('res.partner', string='Transporter Company', required=True, help="The partner that is doing the delivery service.")
+    product_id = fields.Many2one('product.product', string='Delivery Product', required=True)
```

Comment:
do not create a product like this, just do inherits product.template and rest will be manned automatically.